src/main/kotlin/no/nav/familie/ef/sak/vedtak/uttrekk/UttrekkArbeidssøkerRepository.kt

```diff
@@ -0,0 +1,38 @@
+package no.nav.familie.ef.sak.vedtak.uttrekk
+
+import no.nav.familie.ef.sak.repository.InsertUpdateRepository
+import no.nav.familie.ef.sak.repository.RepositoryInterface
+import org.springframework.data.jdbc.repository.query.Query
+import org.springframework.stereotype.Repository
+import java.time.LocalDate
+import java.time.YearMonth
+import java.util.UUID
+
+@Repository
+interface UttrekkArbeidssøkerRepository : RepositoryInterface<UttrekkArbeidssøkere, UUID>,
+                                          InsertUpdateRepository<UttrekkArbeidssøkere> {
+
+    fun findAllByÅrMåned(årMåned: YearMonth): List<UttrekkArbeidssøkere>
+
+    // language=PostgreSQL
+    @Query("""
+        WITH q AS (
+            SELECT b.id behandling_id, b.fagsak_id, ROW_NUMBER() OVER (PARTITION BY b.fagsak_id ORDER BY b.opprettet_tid DESC) rn
+            FROM behandling b
+            JOIN fagsak f ON b.fagsak_id = f.id
+            WHERE
+              b.type != 'BLANKETT'
+              AND b.resultat IN ('OPPHØRT', 'INNVILGET')
+              AND b.status = 'FERDIGSTILT'
+            )
+        SELECT DISTINCT ON (v.behandling_id) -- Trenger ikke å hente samme vedtak flere ganger 
+               q1.behandling_id, q1.fagsak_id, v.behandling_id behandling_id_for_vedtak, v.perioder 
+          FROM q q1
+          JOIN tilkjent_ytelse ty ON ty.behandling_id = q1.behandling_id
+          JOIN andel_tilkjent_ytelse aty ON ty.id = aty.tilkjent_ytelse
+          JOIN vedtak v ON v.behandling_id = aty.kilde_behandling_id
+        WHERE aty.stonad_tom >= :startdato AND aty.stonad_fom <= :sluttdato
+          AND rn = 1
+    """)
+    fun hentArbeidssøkere(startdato: LocalDate, sluttdato: LocalDate): List<ArbeidsssøkereTilUttrekk>
```

Comment:
Henter man egentlig arbeidssøkere her? Ser ut som det ikke gjøres noe filtrering på det i sql'n, men i koden senere, så kanskje denne kan renames?

Reply:
Sant, her hentes vedtaksperioder for gitt måned